sdk/python/kfp/v2/dsl/__init__.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 Google LLC
```

Comment:
Can we avoid copying these files?

If I understand correctly, the issue is that some DSL code calls `kfp.dsl.Pipeline.get_default_pipeline()` and you want to fork the `kfp.dsl.Pipeline` class that mostly belongs to the compiler. Is that correct?
We can try to decouple `kfp.dsl.Pipeline` class, so that it's easier for you to replace.

Reply:
The issue is the code I added in _component_bridge.py could break v1 use cases, for example, we only accept `GCSPath` and `Model` for non-primitive types for now, everything else will raise error. So in order to not break v1 behavior, either I guard my change with some sort of feature flag if that's possible, or I fork the file under a different namespace, i.e. kfp/v2/dsl. (I had to fork dsl/_pipeline.py as well, since it references _component_bridge.py).

Later on, we want to refactor/cleanup the code to make code under v2 contains only IR-related logic.

Reply:
>We can try to decouple kfp.dsl.Pipeline class, so that it's easier for you to replace.

Check the https://github.com/kubeflow/pipelines/pull/4610 PR. It decouples the component_bridge from dsl and moves it to the compiler.
You do not even need to wait for that PR to be merged. Just copy the following changes to your forked compiler and you no longer need to fork anything in DSL (other than the component_bridge). https://github.com/kubeflow/pipelines/pull/4610/files#diff-618bb1fbca23caf0b3edc26f1b38caa8R817 and https://github.com/kubeflow/pipelines/pull/4610/files#diff-618bb1fbca23caf0b3edc26f1b38caa8R1030